loopy/symbolic.py

```diff
@@ -2608,6 +2608,217 @@ def do_access_ranges_overlap_conservative(
 # }}}
 
 
+# {{{ check for write races in accesses
+
+def _check_for_access_races(map_a, insn_a, map_b, insn_b, knl, callables_table):
```

Comment:
I'm not sure `symbolic` is a good place for this. The file is too long as it is. `check` maybe?

Reply:
Agreed that was misplaced, since `loopy.schedule` was the only user, moved it to `loopy.schedule.tools`